File: frontend/src/api/jobs.ts
import { apiClient } from "./client";

export interface JobStep {
  name: string;
  status: "pending" | "running" | "done" | "failed";
  started_at: string | null;
  finished_at: string | null;
  error: string | null;
}

export interface JobStatus {
  job_id: string;
  type: string;
  status: "pending" | "running" | "done" | "failed";
  steps: JobStep[];
  run_id: string | null;
  result: Record<string, unknown> | null;
  error: string | null;
  created_at: string;
  updated_at: string;
}

export async function getJobStatus(jobId: string): Promise<JobStatus> {
  const res = await apiClient.get(`/api/jobs/${jobId}`);
  return res.data;
}

export async function cancelJob(jobId: string): Promise<void> {
  await apiClient.delete(`/api/jobs/${jobId}`);
}

File: backend/app/models/jobs.py
from datetime import datetime
from typing import Any, Optional

from pydantic import BaseModel


class JobStep(BaseModel):
    name: str
    status: str  # pending | running | done | failed
    started_at: Optional[datetime] = None
    finished_at: Optional[datetime] = None
    error: Optional[str] = None


class JobStatus(BaseModel):
    job_id: str
    type: str
    status: str  # pending | running | done | failed
    steps: list[JobStep] = []
    run_id: Optional[str] = None
    result: Optional[dict[str, Any]] = None
    error: Optional[str] = None
    created_at: datetime
    updated_at: datetime
